tree/ntuple/v7/inc/ROOT/libdaos_mock/daos.h

```diff
@@ -239,12 +232,45 @@ int daos_obj_update(daos_handle_t oh, daos_handle_t th, uint64_t flags,
 
 //////////////////////////////////////////////////////////////////////////////// daos_prop.h
 
-
 /** daos properties, for pool or container */
 typedef struct {
 	char		unused; // silence [-Wextern-c-compat]
 } daos_prop_t;
 
+#define DAOS_PROP_LABEL_MAX_LEN (127)
+#define DAOS_PROP_MAX_LABEL_BUF_LEN (DAOS_PROP_LABEL_MAX_LEN + 1)
+
+/**
+ * Check if DAOS (pool or container property) label string is valid.
+ * DAOS labels must consist only of alphanumeric characters, colon ':',
+ * period '.', hyphen '-' or underscore '_', and must be of length
+ * [1 - DAOS_PROP_LABEL_MAX_LEN].
+ * Unlike in libdaos, UUIDs are deemed valid labels for libdaos_mock.
+ * \param[in] label	    Label string
+ * \return	  true      Label meets length/format requirements
+ *			  false     Label is not valid length or format
+ */
+static inline bool daos_label_is_valid(const char *label)
+{
+   int len;
+   int i;
+
+   if (label == NULL)
+      return false;
+
+   len = strnlen(label, DAOS_PROP_LABEL_MAX_LEN + 1);
+   if (len == 0 || len > DAOS_PROP_LABEL_MAX_LEN)
+      return false;
+
+   // Verify that it contains only alphanumeric characters or :.-_
+   for (i = 0; i < len; i++) {
+      char c = label[i];
+      if (isalnum(c) || c == '.' || c == '_' || c == ':' || c == '-')
+         continue;
+      return false;
+   }
+   return true;
```

Comment:
In principle, and to keep the header small, we can even simply `return true;` here, as it is not important for the unit tests.